tests/unit/is_access_granted_ipv4.pl

```diff
@@ -0,0 +1,214 @@
+#! /usr/bin/env perl
+# vim: set filetype=perl ts=4 sw=4 sts=4 et:
+use common::sense;
+use Test::More;
+use Test::Deep;
+
+use File::Basename;
+use lib dirname(__FILE__) . '/../../lib/perl';
+use OVH::Bastion;
+use OVH::Result;
+use JSON;
+
+OVH::Bastion::enable_mocking();
+OVH::Bastion::set_mock_data(
+    {
+        "accounts" => {
+            "me" => {
+                "uid"               => 99982,
+                "gid"               => 99982,
+                "personal_accesses" => [
+                    qw{
+                      me@192.0.2.12:12
+                      me@198.51.100.0/28:12
+                      me@192.0.2.13
+                      me@198.51.100.16/28
+
+                      192.0.2.22:12
+                      198.51.100.32/28:12
+                      192.0.2.23
+                      198.51.100.48/28
+                      }
+                ],
+            },
+        },
+    }
+);
+OVH::Bastion::load_configuration(
+    mock_data => {
+        bastionName => "mock",
+    }
+);
+
+my %want;    # truth table
+my $undef = undef;    # can't use $hash{undef} directly or it means $hash{'undef'}
```

Comment:
This is `my $undef;`, which is also `undef()`.
Though this is wrong, a Perl hash is a `map[string]any`, you cannot put `undef` as a key.

```perl
$ perl -Mstrict -MData::Dumper -E 'my %h = (undef() => 42); defined((keys(%h))[0]) && say "defined"'
defined
```

Reply:
Good catch, replaced it with a special value that'll look for when $port or $user is `undef`